Pod/Classes/VideoComponent/ARVideoManager.m

```diff
@@ -0,0 +1,151 @@
+#import <AVFoundation/AVFoundation.h>
+#import "ARVideoManager.h"
+
+
+@interface ARVideo : UIView
+@property (nonatomic, strong, readwrite) AVPlayer *player;
+@property (nonatomic, strong, readwrite) AVPlayerLayer *playerLayer;
+@end
+
+
+@implementation ARVideo
+
+- (void)setFrame:(CGRect)frame
+{
+    [super setFrame:frame];
+    self.playerLayer.frame = self.bounds;
+}
+
+// Loop
+
+- (void)playerDidReachEnd:(NSNotification *)notification
+{
+    AVPlayerItem *playerItem = [notification object];
+    [playerItem seekToTime:kCMTimeZero];
+    [self.player play];
+}
+
+// We want to know how large the video is to be able to set the player positioning ourselves
+// because we want to have a left-aligned video, instead of the default center aligned fill.
+
+- (void)observeValueForKeyPath:(NSString *)path ofObject:(id)object change:(NSDictionary *)change context:(void *) context
+{
+    // Only do the positioning logic for aspect fills
+    if (self.playerLayer.videoGravity == AVLayerVideoGravityResizeAspectFill && [path isEqualToString:@"readyForDisplay"]) {
+        CGRect videoRect = self.playerLayer.videoRect;
+        
+        if (!CGRectIsEmpty(videoRect)) {
+            // Ok, now we've got enough info to handle the sizing
+            [self updatePlayerLayerFrame];
```

Comment:
So this was basically just a race condition?

Reply:
Yah, it could have been if the moving had been done during the creation of the `playerLayer` - needed the `videoRect` to be certain that we had the `optimumSize` in the video asset inside the AVPlayer